sirplatform.h

```diff
@@ -144,7 +163,9 @@
 #   include <pthread_np.h>
 #   include <sys/sysctl.h>
 #  elif defined(__linux__)
-#   include <linux/limits.h>
+#   if defined(__GLIBC__)
```

Comment:
The order works around a bug in some older headers - it could be avoided but results in some harmless redefintion warning that I found distasteful.

And that header is not available (or never installed) on musl-libc based systems, but the already included limits.h suffices.  It is always available on every glibc-based system I've tested (back to 2012.)

Reply:
I literally do not know what musl is, or half the OSes you mention. I feel like I've been living under a rock. I thought I was being exotic with FreeBSD and arm64.